agenta-web/cypress/e2e/regex-evaluation.cy.ts

```diff
@@ -0,0 +1,148 @@
+
+            cy.clickLinkAndWait('[data-cy="start-new-evaluation-button"]')
+
+            cy.location("pathname").should("include", "/auto_regex_test")
+
+            cy.get(".ant-form-item-explain-error").should("not.exist")
+        })
+
+        it("Should display error for missing regex pattern", () => {
+            cy.clickLinkAndWait('[data-cy="regex-run-evaluation"]')
+
+            cy.get(".ant-form-item-explain-error").should("exist")
+        })
+
+        it("Should select the 'Match' strategy and run the evaluation", () => {
+            cy.get('[data-cy="regex-evaluation-input"]').type(`^[A-Z][a-z]*$`)
+
+            cy.get('[data-cy="regex-evaluation-strategy"]').within(() => {
+                cy.get("label").eq(0).click()
+            })
+
+            cy.clickLinkAndWait('[data-cy="regex-run-evaluation"]')
+
+            cy.get('[data-cy="regex-evaluation-regex-match"]', {timeout: 15000})
+                .invoke("text")
+                .then((text) => {
+                    // Check if the text contains either "Match" or "Mismatch"
+                    expect(text.includes("Match") || text.includes("Mismatch")).to.be.true
+                })
+            cy.get('[data-cy="regex-evaluation-score"]', {timeout: 15000})
+                .invoke("text")
+                .then((text) => {
+                    // Check if the text contains either "correct" or "wrong"
+                    expect(text.includes("correct") || text.includes("wrong")).to.be.true
+                })
+
+            cy.get(".ant-message-notice-content").should("exist")
+            cy.wait(3000)
+            cy.get(".ant-message-notice-content").should("not.exist")
+        })
+
+        it("Should select the 'Mismatch' strategy and run the evaluation", () => {
+            cy.get('[data-cy="regex-evaluation-input"]').type(`^[A-Z][a-z]*$`)
+
+            cy.get('[data-cy="regex-evaluation-strategy"]').within(() => {
+                cy.get("label").eq(1).click()
+            })
+
+            cy.clickLinkAndWait('[data-cy="regex-run-evaluation"]')
+
+            cy.get('[data-cy="regex-evaluation-regex-match"]', {timeout: 15000})
+                .invoke("text")
+                .then((text) => {
+                    // Check if the text contains either "Match" or "Mismatch"
+                    expect(text.includes("Match") || text.includes("Mismatch")).to.be.true
+                })
+            cy.get('[data-cy="regex-evaluation-score"]', {timeout: 15000})
+                .invoke("text")
+                .then((text) => {
+                    // Check if the text contains either "correct" or "wrong"
+                    expect(text.includes("correct") || text.includes("wrong")).to.be.true
+                })
+            cy.get(".ant-message-notice-content").should("exist")
+            cy.wait(3000)
+            cy.get(".ant-message-notice-content").should("not.exist")
+        })
+    })
+
+    context("Cleanup", () => {
```

Comment:
Cleanup should not be a test case. Please perform the cleanup inside after() clause